metpy/calc/tests/test_basic.py

```diff
@@ -264,6 +264,16 @@ def test_heat_index_ratio():
     assert_almost_equal(hi.to('degC'), units.Quantity([50.3405, np.nan], units.degC), 4)
 
 
+def test_heat_index_vs_nws():
+    """Test heat_index against online calculated HI from NWS Website."""
+    # https://www.wpc.ncep.noaa.gov/html/heatindex.shtml, visited 2019-Jul-17
+    temp = units.Quantity(np.array([86, 111, 40, 96]), units.degF)
```

Comment:
So I take it this test fails if we remove the 40F threshold?

Reply:
So if I comment that `40F` logic out, we get this:

```
>>> from metpy.calc.basic import heat_index
>>> from metpy.units import units
>>> heat_index(40. * units.degF, 50 * units.percent, mask_undefined=False)
<Quantity([36.05], 'degF')>
>>> heat_index(40. * units.degF, 50 * units.percent, mask_undefined=True)
<Quantity([--], 'degF')>
```

Reply:
Sigh, nevermind, I proved your point.

Reply:
So this isn't good behavior then.  We are getting bailed out by `apparent_temperature` using the windchill mask for its logic.  If that was reversed, we'd get explosions.   UFFFF.  It needs removed then, right?

Reply:
Sorry, I should go to bed.  We are still good.  This test should fail if there wasn't a bug with nan checking?

Reply:
What is this sorcery?
```
import numpy.testing
import numpy as np
# does not raise AssertionError
numpy.testing.assert_array_almost_equal(
  np.array([10, 20]), np.ma.array([10, np.nan], mask=[False, True]), 2)
```

`metpy.testing.assert_array_almost_equal` is not doing the right thing then?

Reply:
There may be some weirdness here, but let's address that elsewhere. I want to get this merged because I have to add a conflicting change to the Travis config to fix being broken by the latest docutils. :scream: